src/Microsoft.Azure.WebJobs.Host/Timers/TaskSeriesTimer.cs

```diff
@@ -147,9 +147,9 @@ private async Task RunAsync(CancellationToken cancellationToken)
                         TaskSeriesCommandResult result = await _command.ExecuteAsync(cancellationToken);
                         wait = result.Wait;
                     }
-                    catch (Exception ex) when (ex.InnerException is TaskCanceledException)
+                    catch (Exception ex) when (ex.InnerException is OperationCanceledException)
```

Comment:
Took me a while to realize that moving this to `OperationCanceledException` did everything I needed. The new behavior throws an `OperationCanceledException` wrapped in a `StorageException` if the network hangs for 100 seconds. Now, that exception will bubble up to here and the loop will start over, just as we'd want it to. (and it behaves the same as before as `TaskCanceledException : OperationCanceledException`).